### DISCUSSION TITLE
Terser Function Declaration Syntax without `()`

### DISCUSSION THREAD

**Discussion by msadeqhe** [hsutter/cppfront#806] [Suggestions]
Consider this function declaration syntax:

```cpp
fnc1: () -> i32 = 10;
```

What's your opinion if we could drop `()` from it?

```cpp
fnc1: -> i32 = 10;
```

And in lambda:

```cpp
: -> i32 = 10
```

Finally, it would be:

```cpp
a: () -> i32 = 10;
b: -> i32 = 10;
c: i32 = 10;
```

`a` and `b` are functions, but `c` is a variable.

**Comment by msadeqhe** [comment]
BTW, I think because `()` is required to call functions, it seems reasonable to require `()` in function declarations too. In other words, the current syntax looks good in which `()` is required to declare functions even if the parameter list is empty.

**Reply by AbhinavK00** [reply]
I agree with this, `()` screams a function to me so it may as well be there for all kinds of functions.
